File: Javascript/3.js
#!/usr/bin/env node

/* 
Project Euler Solutions
Problem 3
*/

var euler = require('./euler.js');

var input = 600851475143;
//var input = 5001;
var factors = euler.getFactors(input);
console.log(factors);


var primeFactors = [];

for (j = 0; j < factors.length; j++) {
  if (euler.isPrime(factors[j])) {
    primeFactors.push(factors[j]);
  } 
}

console.log(primeFactors);
console.log("Max prime factor: " + euler.max(primeFactors).toString());
console.log("Min prime factor: " + euler.min(primeFactors).toString());

File: c/3.c
// The prime factors of 13195 are 5, 7, 13 and 29.

// What is the largest prime factor of the number 600851475143 ?

// Answer:  6857

// input type = unsigned long long?
// math.sqrt
// isFactor
// isPrime
#include <stdio.h>
#include "euler.h"

int main(int argc, char const *argv[])
{
    long long in = 600851475143;
    unsigned int i;
    for (i = in - 2; i > 3; i -= 2)
    {
        if (isFactor(i, in))
        {
            if (isPrime(i))
            {
                printf("%d", i);
                return 0;
            }
        }
    }
    return -1;
}
